AttitudeControl: let the user set yaw weight directly via parameter

MaEtUgR opened the pull request:
instead of infering the yaw weight from the gains which can lead to
unexpected results depending on the particular vehicle tuning.

jkflying commented:
This is setting a default of 0.4, what would the defaults have roughly been before? I don't have a feeling for what yaw P gain would be compared to pitch/roll.

MaEtUgR commented:
@jkflying Good point. I should mention:
- The default gain parameters of PX4 made almost exactly a ratio of 0.4.
- The paper https://www.research-collection.ethz.ch/bitstream/handle/20.500.11850/154099/eth-7387-01.pdf under chapter 4.2 states that their experimental results were best with 0.4.
- During my test flying for my master thesis I also experimented with this weight and around 0.4 gave me the most desirable results.
- I made a prototyping script in python with visualization here https://github.com/Auterion/Flight_Control_Prototyping_Scripts/tree/master/quaternion_attitude_control#details in which you can try yourself what value results in which attitude run.

sfuhrer commented:
@MaEtUgR this is worth keeping open, right?

MaEtUgR commented:
@sfuhrer Well yes, I should push it through and do the necessary testing 😬 
It got less urgent since we have yaw slew rates in missions but it's not a final solution because there's also no feed-forward.

MaEtUgR commented:
I revived this and finished the work:
- rebased on master (first commit was existing)
- fixed the yaw output scaling which was like expected broken with the original first draft
- Added unit tests to verify that the yaw output scaling is correct and doesn't result in division by zero

MaEtUgR commented:
I found a better implementation for the exact same thing and squashed my commits.

MaEtUgR commented:
@PX4/testflights Could you please do a quick test on this with focus on multicopter yawing? Stabilized, Altitude, Position, Mission just check if yaw tracking looks like expected.

Junkim3DR commented:
### Tested NXP FMUK66 V3

**Modes Tested**
- Position Mode: Good.
- Altitude Mode: Good.
- Stabilized Mode: Good.
- Mission Plan Mode (Automated): Good.
- RTL (Return To Land): Good.

**Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint then trigger RTL.

**Notes**
No issues noted, good flight in general.

**Logs**
- https://review.px4.io/plot_app?log=2b83350f-df29-4956-8fd8-7b226b6c9923
- https://review.px4.io/plot_app?log=1dd134b3-b12a-448d-a2bf-3e6d6e7524f5


### Tested Pixhawk 3 V4Pro

**Modes Tested**
- Position Mode: Good.
- Altitude Mode: Good.
- Stabilized Mode: Good.
- Mission Plan Mode (Automated): Good.
- RTL (Return To Land): Good.

**Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint then trigger RTL.

**Notes**
Vehicle does not detect all Flight Modes Switches, Top = Stabilized ; Mid = Altitude (Undetected No Flight Mode Change) ; Bot = Position (Switched to altitud) RC Work with no issue on other vehicles.

**Logs**
- https://review.px4.io/plot_app?log=b28b4721-682c-47ce-a8f9-1d47740b2756

dannyfpv commented:
Tested on pixhawk4 v5 f-450
Modes Tested

Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL (Return To Land): Good.
Procedure
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint then trigger RTL.

Notes
No issues noted, good flight in general.

Logs
https://review.px4.io/plot_app?log=aa76011d-28d7-413b-81b5-3fb3047e9b65
https://review.px4.io/plot_app?log=f85caf4c-973d-43d9-9904-640ddcfb237a

jorge789 commented:
### Tested on PixRacer V4

**Modes Tested**

Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL (Return To Land): Good.
**Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint then trigger RTL.

**Notes**
The following issue was not seen in this PR _"In altitude mode, the vehicle did not respect the height that continued to increase in pitch forward"_
We are unsure if this was due to the lack of strong winds on the field today.


**Logs**
https://review.px4.io/plot_app?log=54f0c926-14eb-47de-a996-6a0ee3f33710

https://review.px4.io/plot_app?log=f667ae28-5495-4694-ae36-b85ddff54698

### Tested on CUAV nano V5

**Modes Tested**

Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL (Return To Land): Good.
**Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint then trigger RTL.

**Notes**
No issues noted, good flight in general.

**Logs**
https://review.px4.io/plot_app?log=4f4c2181-b7ea-4707-b46b-345b58c3c0ca

https://review.px4.io/plot_app?log=cd036b34-a4c4-4764-af43-1277d34552b4

MaEtUgR commented:
Thanks so much for testing!
